rl_chain/slates_chain.py

```diff
@@ -101,70 +97,45 @@ def to_vw_format(
             ]
         )
         ps = (
-            [f"{a}:{p}" for a, p in slates_label.get_actions_and_probs()]
-            if slates_label
+            [f"{a}:{p}" for a, p in vw_event.label.get_actions_and_probs()]
+            if vw_event.label
             else [""] * len(action_features)
         )
         slots = [f"slates slot {p} |" for p in ps]
         return "\n".join(list(chain.from_iterable([[context_str], actions, slots])))
 
 
-class Policy(ABC):
-    @abstractmethod
-    def predict(
-        self, inputs: Dict[str, Any], actions: Dict[str, Any], context: Dict[str, Any]
-    ) -> Label:
-        ...
+class RandomPolicy(base.Policy):
+    def __init__(self, text_embedder: base.Embedder, *_, **__):
+        self.text_embedder = text_embedder
 
+    def predict(self, event: SlatesPersonalizerChain.Event) -> Any:
+        return [
+            [(random.randint(0, len(slot) - 1), 1.0 / len(slot))]
+            for slot in self.text_embedder.to_action_features(event.actions)
+        ]
 
-class VwPolicy(Policy):
-    def __init__(
-        self, workspace: vw.Workspace, text_embedder: SlatesTextEmbedder, *_, **__
-    ):
-        self.workspace = workspace
-        self.text_embedder = text_embedder
+    def learn(self, event: SlatesPersonalizerChain.Event) -> Any:
+        pass
 
-    def predict(
-        self, inputs: Dict[str, Any], actions: Dict[str, Any], context: Dict[str, Any]
-    ) -> Label:
-        text_parser = vw.TextFormatParser(self.workspace)
-        return Label(
-            self.workspace.predict_one(
-                base.parse_lines(
-                    text_parser,
-                    self.text_embedder.to_vw_format(
-                        inputs=inputs, actions=actions, context=context
-                    ),
-                )
-            )
-        )
+    def log(self, event: SlatesPersonalizerChain.Event) -> Any:
+        pass
 
 
-class RandomPolicy(Policy):
-    def __init__(self, text_embedder: SlatesTextEmbedder, *_, **__):
+class FirstChoicePolicy(base.Policy):
+    def __init__(self, text_embedder: base.Embedder, *_, **__):
         self.text_embedder = text_embedder
 
-    def predict(
-        self, inputs: Dict[str, Any], actions: Dict[str, Any], context: Dict[str, Any]
-    ) -> Label:
-        return Label(
-            [
-                [(random.randint(0, len(slot) - 1), 1.0 / len(slot))]
-                for slot in self.text_embedder.to_action_features(actions)
-            ]
-        )
-
+    def predict(self, event: SlatesPersonalizerChain.Event) -> Any:
+        return [
+            [(0, 1)] for slot in self.text_embedder.to_action_features(event.actions)
```

Comment:
```suggestion
            [(0, 1)] for slot in event.actions
```

Reply:
the Event stores the actions from the input, not the featurized version of them so we need to call to_action_features for the policies here

Reply:
but these specific implementations does not care about features, only about #actions or their order?

Reply:
ah yes true

Reply:
fixed